### ISSUE TITLE
[CI] IndexTemplateRegistryTests.testRolloverForFreshInstalledIndexTemplate

### ISSUE THREAD

**Issue by thecoop** [elastic/elasticsearch#116414]
### CI Link

https://gradle-enterprise.elastic.co/s/s3twsibxo5wrc

### Repro line

`./gradlew ":x-pack:plugin:core:test" --tests "org.elasticsearch.xpack.core.template.IndexTemplateRegistryTests.testRolloverForFreshInstalledIndexTemplate" -Dtests.seed=99F58F8705D188EC -Dtests.locale=dje-NE -Dtests.timezone=Asia/Istanbul -Druntime.java=23`

### Does it reproduce?

No

### Applicable branches

main

### Failure excerpt

```
IndexTemplateRegistryTests > testRolloverForFreshInstalledIndexTemplate FAILED
    java.lang.AssertionError: 
    Expected: <2>
         but: was <0>
        at __randomizedtesting.SeedInfo.seed([99F58F8705D188EC:A88798C75D4DDE3C]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
        at org.elasticsearch.test.ESTestCase.assertThat(ESTestCase.java:2543)
        at org.elasticsearch.xpack.core.template.IndexTemplateRegistryTests.testRolloverForFreshInstalledIndexTemplate(IndexTemplateRegistryTests.java:479)
```

**Comment by dakrone** [comment]
@axw I believe this is related to https://github.com/elastic/elasticsearch/pull/116219

**Comment by axw** [comment]
@dakrone sure is, sorry about that - fixed by https://github.com/elastic/elasticsearch/pull/116450

**Comment by dakrone** [comment]
Resolved by #116450
